MathlibTest/grind/grobner.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2022 Dhruv Bhatia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dhruv Bhatia, Robert Y. Lewis, Mario Carneiro, Kim Morrison
+-/
+import Mathlib
+
+/-!
+Mathlib used to have a tactic (`polyrith`) that communicated with a remote Sage server
+to do Grobner basis calculations.
+This test file is adapted from the test file for `polyrith`,
+but uses the `grobner` tactic instead.
+(Recall `grobner` is a thin wrapper around `grind` disabling other modules.)
+-/
+
+/-! ### Standard Cases over ℤ, ℚ, and ℝ -/
+
+example (x y : ℤ) (h1 : 3*x + 2*y = 10) :
+  3*x + 2*y = 10 := by grobner
+
+example (x y : ℚ) (h1 : x*y + 2*x = 1) (h2 : x = y) :
+  x*y = -2*y + 1 := by grobner
+
+example (x y : ℝ) (h1 : x + 2 = -3) (h2 : y = 10) :
+  -y + 2*x + 4 = -16 := by grobner
+
+example (x y z : ℝ) (ha : x + 2*y - z = 4) (hb : 2*x + y + z = -2)
+    (hc : x + 2*y + z = 2) :
+  -3*x - 3*y - 4*z = 2 := by grobner
+
+-- `grind` does not yet understand scientific notation:
+-- example (w x y z : ℝ) (h1 : x + 2.1*y + 2*z = 2) (h2 : x + 8*z + 5*w = -6.5)
+--     (h3 : x + y + 5*z + 5*w = 3) :
+--   x + 2.2*y + 2*z - 5*w = -8.5 := by grobner
```

Comment:
If we use `fail_if_success`, we get reminded to update the test when `grobner` starts working on scientific notation.
```suggestion
example (w x y z : ℝ) (h1 : x + 2.1*y + 2*z = 2) (h2 : x + 8*z + 5*w = -6.5)
    (h3 : x + y + 5*z + 5*w = 3) :
    x + 2.2*y + 2*z - 5*w = -8.5 := by
  -- `grind` does not yet understand scientific notation:
  fail_if_success grobner
  sorry
```